### PR TITLE
[SINT-1438] Add Clipboard Paste Rule to Guarddog

### PR DESCRIPTION
**Pull request by Claire-Thib**
In response to [this](https://github.com/DataDog/guarddog/issues/231), added a semgrep rule that detects when packages use `pyperclip.paste()`

Opted to simply detect `pyperclip.paste()` versus `pyperclip.paste()` and searching the output because the benchmark tests showed that just detecting `pyperclip.paste()` had only **one false positive** and **zero false negatives**.

A test was also added for this rule

### CODE REVIEW

File: guarddog/analyzer/sourcecode/clipboard-paste.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+rules:
+  - id: clipboard-paste
```
Comment (christophetd) [inline guarddog/analyzer/sourcecode/clipboard-paste.yml:2]:
```suggestion
  - id: clipboard-access
```

so we could also cover stealing data from the clipboard in the future?

File: guarddog/analyzer/sourcecode/clipboard-paste.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    languages:
+      - python
+    message: This package has a potential misuse of pyperclip.paste()
+    metadata:
+      description: Identify when the 'pyperclip.paste()' command is executed
```
Comment (christophetd) [inline guarddog/analyzer/sourcecode/clipboard-paste.yml:7]:
```suggestion
      description: Identify when a package reads or write data from the clipboard 
```

File: guarddog/analyzer/sourcecode/clipboard-paste.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+rules:
+  - id: clipboard-paste
+    languages:
+      - python
+    message: This package has a potential misuse of pyperclip.paste()
```
Comment (christophetd) [inline guarddog/analyzer/sourcecode/clipboard-paste.yml:5]:
```suggestion
    message: This package accesses the clipboard, which can indicate it's trying to steal sensitive data or overwrite cryptocurrency addresses
```

File: tests/analyzer/sourcecode/clipboard-paste.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+""" RULEID: KEWK malware
+"""
+# ruleid: clipboard-paste
+clipboard_data = pyperclip.paste()
+if re.search('^[13][a-km-zA-HJ-NP-21-9]{25, 34}$', clipboard_data):
+    if clipboard_data not in [self.address_btc, self.address_eth, self.address_xchain]:
+        if self.address_btc != "none":
+            pyperclip.copy(self.address_btc)
+            # ruleid: clipboard-paste
+            pyperclip.paste()
```
Comment (christophetd) [inline tests/analyzer/sourcecode/clipboard-paste.py:17]: any clue why the malware uses pyperclip.paste() here?

File: guarddog/analyzer/sourcecode/clipboard-paste.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+    metadata:
+      description: Identify when the 'pyperclip.paste()' command is executed
+    patterns:
+      - pattern: pyperclip.paste()
```
Comment (christophetd) [inline guarddog/analyzer/sourcecode/clipboard-paste.yml:9]: if I understand correctly, `pyperclip.paste()` is accessing data in the keyboard - if that's the case, shouldn't we also match on `pyperclip.copy` (writing data to it)?

File: tests/analyzer/sourcecode/clipboard-paste.py (hunk 4 above)
Comment (Claire-Thib) [reply tests/analyzer/sourcecode/clipboard-paste.py:17]: Oh I don't think it has any impact. I'll remove it

**Pull request merged**